src/main/kotlin/com/codecraft/ai/api/engine/KeyFusionAPI.kt

```diff
@@ -0,0 +1,68 @@
+package com.codecraft.ai.api.engine
+
+import com.codecraft.ai.api.models.Text2ImageRequest
+import com.codecraft.ai.api.models.Text2ImageResponse
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import kotlinx.serialization.json.Json
+import java.nio.charset.StandardCharsets
+import java.util.*
+
+suspend fun startText2ImageProcess(call: ApplicationCall) {
+    call.request.apply {
+        val promptFromUser = queryParameters[QueryParam.PROMPT.strRepresentation] ?: return Error.EMPTY_PROMPT.let {
+            call.respond(
+                HttpStatusCode(
+                    it.code,
+                    it.strRepresentation
+                )
+            )
+        }
+
+        connectToSDAndGetImage(
+            Text2ImageRequest(
+                prompt = promptFromUser,
+                steps = queryParameters[QueryParam.STEPS.strRepresentation]?.toInt() ?: 25,
+                width = queryParameters[QueryParam.WIDTH.strRepresentation]?.toInt() ?: 512,
+                height = queryParameters[QueryParam.HEIGHT.strRepresentation]?.toInt() ?: 512,
+            ),
+        )?.let {
+            call.respond(it)
+        }
+    }
+}
+
+suspend fun connectToSDAndGetImage(text2ImageRequest: Text2ImageRequest): ByteArray? {
+    val client = HttpClient(CIO) {
+        install(HttpTimeout) {
+            connectTimeoutMillis = HttpTimeout.INFINITE_TIMEOUT_MS
+            requestTimeoutMillis = HttpTimeout.INFINITE_TIMEOUT_MS
+            socketTimeoutMillis = HttpTimeout.INFINITE_TIMEOUT_MS
+        }
+
+        install(ContentNegotiation) {
+            json(Json {
+                prettyPrint = true
+                ignoreUnknownKeys = true
+            })
+        }
+    }
+
+    val response: Text2ImageResponse = client.post(StableDiffusionLocalhost) {
+        contentType(ContentType.Application.Json)
+        setBody(text2ImageRequest)
+    }.body()
+
+    response.images.firstOrNull()?.let {
+        return Base64.getDecoder().decode(it.toByteArray(StandardCharsets.UTF_8))
+    }
+    return null
```

Comment:
I think its better to throw exception than return null here

Reply:
The caller function will get this null and will send an error code to the user, so no need to throw an exception and then catch in the caller function and send an error code based on this exception, will be boilerplate

Reply:
is not it worse that we have nulls everywhere and then we have to use `.?let`